multiplexer/abci/remote_v1.go

```diff
@@ -0,0 +1,636 @@
+package abci
+
+import (
+	"context"
+	"math"
+
+	"google.golang.org/grpc"
+
+	abciv2 "github.com/cometbft/cometbft/abci/types"
+	cryptov2 "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	typesv2 "github.com/cometbft/cometbft/proto/tendermint/types"
+	abciv1 "github.com/tendermint/tendermint/abci/types"
+	cryptov1 "github.com/tendermint/tendermint/proto/tendermint/crypto"
+	typesv1 "github.com/tendermint/tendermint/proto/tendermint/types"
+	versionv1 "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+type RemoteABCIClientV1 struct {
+	abciv1.ABCIApplicationClient
+
+	// retainLastHeight is the height is set in finalize block
+	// and returned in commit
+	commitRetainLastHeight int64
+	// endBlockConsensusVersion is the app version got from the end block abci call
+	endBlockConsensusAppVersion uint64
+	// chainID is required to pass into the headers.
+	chainID string
+}
+
+// NewRemoteABCIClientV1 returns a new ABCI Client (using ABCI v1).
+// The client behaves like Tendermint for the server side (the application side).
+func NewRemoteABCIClientV1(conn *grpc.ClientConn, chainID string) *RemoteABCIClientV1 {
+	return &RemoteABCIClientV1{
+		ABCIApplicationClient: abciv1.NewABCIApplicationClient(conn),
+		chainID:               chainID,
+	}
+}
+
+// ApplySnapshotChunk implements abciv2.ABCI
```

Comment:
[question] why does remote abci client v1 accept requests of type abciv2 and return responses of type abciv2?